Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPrinting.mm

```diff
@@ -86,3 +86,63 @@ - (void)waitForPagination
     [webView evaluateJavaScript:@"window.print()" completionHandler:nil];
     [delegate waitForPagination];
 }
+
+#if PLATFORM(MAC)
+@interface WKPrintPageBordersWebView: TestWKWebView
+@end
+
+@implementation WKPrintPageBordersWebView {
+    bool _didDrawPageBorder;
+}
+
+- (void)drawPageBorderWithSize:(NSSize)borderSize
+{
+    _didDrawPageBorder = true;
+}
+
+- (void)_waitUntilPageBorderDrawn
+{
+    TestWebKitAPI::Util::run(&_didDrawPageBorder);
+}
+
+@end
+
+@interface PrintShowingPrintPanelUIDelegate : NSObject <WKUIDelegate>
+
+@end
+
+@implementation PrintShowingPrintPanelUIDelegate
+
+- (void)_webView:(WKWebView *)webView printFrame:(_WKFrameHandle *)frame pdfFirstPageSize:(CGSize)size completionHandler:(void (^)(void))completionHandler
+{
+    NSPrintInfo *printInfo = [[NSPrintInfo alloc] init];
```

Comment:
This technically leaks. It's fine since we run tests singly, but would be nice if you'd adopt it into a RetainPtr instead.